metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java

```diff
@@ -47,7 +46,6 @@
       Deserializer s = HiveMetaStoreUtils.getDeserializer(conf, tbl, false);
       return HiveMetaStoreUtils.getFieldsFromDeserializer(tbl.getTableName(), s);
     } catch (Exception e) {
-      StringUtils.stringifyException(e);
```

Comment:
MetaException is a different animal. When we send it through thrift we lose the inner exceptions.
Could you please check if we do not lose "caused by" info in HS2 logs after this change?